src/include/dns-server/dialog-main.rb

```diff
@@ -1898,12 +1898,36 @@ def SaveAndRestart(event)
 
     # Writes DNS server settings and saves the service
     #
-    # @note currently, the DnsServer is a Perl module, reason why the write of
-    # settings is being performed in two steps.
+    # As side effect, it also sets/resets the local forwarder
     #
     # @return [Boolean] true if settings are saved successfully; false otherwise
     def write_settings
-      DnsServer.Write && service.save
+      return false unless DnsServer.Write
+
+      performed_action = service.action
+      service_saved = service.save(keep_state: Mode.auto)
+
+      reset_local_forwarder(performed_action, service_saved)
+
+      service_saved
+    end
+
+    # Resets the local forwarder to "resolver"
+    #
+    # Local forwarder will be changed to "resolver" if
+    #
+    #   * its current value is not "bind"
+    #   * service was stopped
+    #   * service cannot be started, restarted or reloaded
+    #
+    # @param performed_action [Symbol] action executed
+    # @param service_saved [Boolean] result of peforms the service#save
+    def reset_local_forwarder(performed_action, service_saved)
+      return unless DnsServer.GetLocalForwarder == "bind"
+      return unless !service_saved || performed_action == :stop
```

Comment:
Negating a condition within an unless might be hard to read. Applying one of the De Morgan's law, you can write this (which is more readable to me):

```ruby 
return if service_saved && performed_action != :stop
```